[nrf fromtree] net: ip: net_context: AF_PACKET/SOCK_RAW/IPPROTO_RAW: set pkt family

jhirsi opened the pull request:
Pull PR: https://github.com/zephyrproject-rtos/zephyr/pull/53440 into the NCS.

rlubos commented:
Will need a manifest update in `sdk-nrf` for this (I suggest to wait for the upmerge to finish though).

desowin reviewed (changes requested):
This is not fremtree! Upstream SHA is e92b067b7fdcf36032d2bfbd06c4dc20164acf83

jhirsi commented:
> This is not fromtree, but rather fromlist! This is merged and therefore to make it fromtree, the upstream SHA [e92b067](https://github.com/nrfconnect/sdk-zephyr/commit/e92b067b7fdcf36032d2bfbd06c4dc20164acf83) has to be used while cherry-picking.

Rebased and updated a SHA in a commit msg.
